bucket/vscode-insiders.json

```
@@ -49,7 +49,21 @@
     "persist": "data",
     "checkver": {
         "url": "https://update.code.visualstudio.com/api/update/win32-x64-archive/insider/latest",
-        "jsonpath": "$.productVersion"
+        "jsonpath": "$['productVersion','timestamp']",
+        "script": [
+            "# Simulate scoop's JsonPath behavior to work around a bug in Newtonsoft.Json's",
+            "# support for JsonPath's multiple property union",
+            "Add-Type -Path \"$script:PSScriptRoot\\..\\supporting\\validator\\bin\\Newtonsoft.Json.dll\"",
+            "$obj = [Newtonsoft.Json.JsonConvert]::DeserializeObject($page)",
+            "# This is the only change: change errorWhenNoMatch argument from $true to $false",
+            "# to work around the bug",
+            "$result = $obj.SelectTokens($jsonpath, $false)",
+            "# remove $jsonpath variable from parent scope to prevent Scoop's logic from running",
+            "$script:jsonpath = $null",
+            "$([String]::Join('\\n', $result))"
+        ],
```

```suggestion
        "jsonpath": "$..['productVersion','timestamp']",
```
This seems to work without the workaround

---

Fair point @kazzarin, that's a much simpler workaround!

I'm not a huge fan of the implicit nature of recursive descent but it's certainly a much simpler way to deal with this until the Newtonsoft.Json bug is fixed!